scipy/stats/_binned_statistic.py

```diff
@@ -664,3 +632,70 @@ def binned_statistic_dd(sample, values, statistic='mean',
     result = result.reshape(input_shape[:-1] + list(nbin-2))
 
     return BinnedStatisticddResult(result, edges, binnumbers)
+
+
+def _bin_edges(sample, bins=None, range=None):
```

Comment:
even though the function is 'private' there should still be a docstring, it makes maintenance and future debugging a lot easier.

Reply:
I added a short description of the private functions but I think the improved modularity already makes maintenance easier.

Reply:
@egouden I agree the added modularity helps. Docstrings are helpful for when you come back to the function after several months/years and can't remember the specifics.